sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/TranslationStatusResult.java

```diff
@@ -0,0 +1,203 @@
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Translation job status response.
+ */
+@Immutable
+public final class TranslationStatusResult implements JsonSerializable<TranslationStatusResult> {
+
+    /*
+     * Id of the translation operation.
+     */
+    @Generated
+    private final String id;
+
+    /*
+     * Operation created date time
+     */
+    @Generated
+    private final OffsetDateTime createdOn;
+
+    /*
+     * Date time in which the operation's status has been updated
+     */
+    @Generated
+    private final OffsetDateTime lastUpdatedOn;
+
+    /*
+     * List of possible statuses for job or document
+     */
+    @Generated
+    private final TranslationStatus status;
+
+    /*
+     * This contains an outer error with error code, message, details, target and an
+     * inner error with more descriptive details.
+     */
+    @Generated
+    private TranslationError error;
+
+    /*
+     * Status Summary
+     */
+    @Generated
+    private final TranslationStatusSummary summary;
+
+    /**
+     * Creates an instance of TranslationStatusResult class.
+     *
+     * @param id the id value to set.
+     * @param createdOn the createdOn value to set.
+     * @param lastUpdatedOn the lastUpdatedOn value to set.
+     * @param status the status value to set.
+     * @param summary the summary value to set.
+     */
+    @Generated
+    private TranslationStatusResult(String id, OffsetDateTime createdOn, OffsetDateTime lastUpdatedOn,
+        TranslationStatus status, TranslationStatusSummary summary) {
+        this.id = id;
+        this.createdOn = createdOn;
+        this.lastUpdatedOn = lastUpdatedOn;
+        this.status = status;
+        this.summary = summary;
+    }
+
+    /**
+     * Get the id property: Id of the translation operation.
+     *
+     * @return the id value.
+     */
+    @Generated
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Get the createdOn property: Operation created date time.
+     *
+     * @return the createdOn value.
+     */
+    @Generated
+    public OffsetDateTime getCreatedOn() {
+        return this.createdOn;
+    }
+
+    /**
+     * Get the lastUpdatedOn property: Date time in which the operation's status has been updated.
+     *
+     * @return the lastUpdatedOn value.
+     */
+    @Generated
+    public OffsetDateTime getLastUpdatedOn() {
+        return this.lastUpdatedOn;
+    }
+
+    /**
+     * Get the status property: List of possible statuses for job or document.
+     *
+     * @return the status value.
+     */
+    @Generated
+    public TranslationStatus getStatus() {
+        return this.status;
+    }
+
+    /**
+     * Get the ResponseError property: This contains error code and message.
+     *
+     * @return the ResponseError value.
+     */
+    public ResponseError getError() {
+        return new ResponseError(this.error.getCode().toString(), this.error.getMessage());
```

Comment:
Same here - check for null and also create a single instance of ResponseError.

Reply:
Updated